integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/PushSmokeTest.java

```diff
@@ -0,0 +1,51 @@
+package com.vaadin.flow.quarkus.it;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+
+import com.vaadin.flow.test.AbstractChromeTest;
+import com.vaadin.testbench.TestBenchElement;
+
+@QuarkusIntegrationTest
+public class PushSmokeTest extends AbstractChromeTest {
+
+    protected String getTestPath() {
+        return "/push";
+    }
+
+    @Test
+    public void pushUpdatesEmbeddedWebComponent() {
+        open();
+        waitForElementPresent(By.id("push-update"));
+
+        int expectedUpdates = 50;
+        int initialUpdateCount = getUpdateCount();
+        Assertions.assertTrue(initialUpdateCount < expectedUpdates,
+                "The initial update count should be less than maximum 50, but it has value "
+                        + initialUpdateCount);
+
+        waitUntil(driver -> getUpdateCount() > initialUpdateCount, 10);
+
+        int nextUpdateCount = getUpdateCount();
+
+        Assertions.assertTrue(nextUpdateCount < expectedUpdates,
+                "The next interim update count should be less than maximum 50, but it has value "
+                        + nextUpdateCount);
+
+        waitUntil(driver -> getUpdateCount() == expectedUpdates, 5);
+
+        int updateCount = getUpdateCount();
+        Assertions.assertEquals(expectedUpdates, updateCount,
+                "The update count should have reached the maximin 50, but it has value "
+                        + updateCount);
```

Comment:
Will the `waitUntil` throw an exception if this condition doesn't get true in the first place, so that if the flow reaches here the assertion can never be false.

Reply:
You're right, the count will never be more than the expected value, so no need for another check after `waitUntil`.
I'll remove it. Thanks